autoload/go/guru.vim

```diff
@@ -198,23 +197,26 @@ function! s:async_guru(args) abort
   endfunction
 
   let start_options = {
-        \ 'close_cb': function("s:close_cb"),
+        \ 'close_cb': funcref("s:close_cb"),
+        \ 'in_mode': 'raw',
+        \ 'out_mode': 'nl',
+        \ 'err_mode': 'nl',
         \ }
 
-  if &modified
-    let l:tmpname = tempname()
-    call writefile(split(result.stdin_content, "\n"), l:tmpname, "b")
-    let l:start_options.in_io = "file"
-    let l:start_options.in_name = l:tmpname
-  endif
-
```

Comment:
Thanks for the work. But I've didn't understand why you're removing this piece. Because that's the only way to pass the stdin content. Seems like you changed it below, which starts first the job and then sends the content of the stdin. Why do we need this? I've read your PR comment but couldn't understand it unfortunately.